src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -197,52 +193,39 @@ private async Task<ScriptSecrets> ReadFunctionSecrets(string functionName)
                 Keys = new List<Key>()
             };
 
-            foreach (Task<SecretBundle> task in tasks)
+            foreach (Task<Response<KeyVaultSecret>> task in tasks)
             {
-                SecretBundle item = task.Result;
-                functionSecrets.Keys.Add(SecretBundleToKey(item, prefix));
+                KeyVaultSecret item = task.Result;
+                functionSecrets.Keys.Add(KeyVaultSecretToKey(item, prefix));
             }
 
             return functionSecrets;
         }
 
-        public static async Task<List<IEnumerable<SecretItem>>> GetKeyVaultSecretsPagesAsync(KeyVaultClient keyVaultClient, string keyVaultBaseUrl)
+        public static AsyncPageable<SecretProperties> GetKeyVaultSecretsPagesAsync(SecretClient keyVaultClient)
```

Comment:
This one I have been debating a bit. I think taking advantage of the new mechanism with IAsyncEnumerable is the right way to go, but I want to make sure it doesn't have any serious impact to move away from `List<IEnumerable`